python3Packages.scancode-toolkit: pre-generate index file

jopejoe1 opened the pull request:
This makes it possible to use the included cli to scan a repo for licenses.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ThanePatrol commented on pkgs/development/python-modules/scancode-toolkit/default.nix:
```diff
@@ -137,9 +137,9 @@ buildPythonPackage rec {
 
-  # Importing scancode needs a writeable home, and preCheck happens in between
-  # pythonImportsCheckPhase and pytestCheckPhase.
+  # Pre-genrating license index (needed as its located in the nix store)
```
nit: remove "(needed as its located in the nix store)"

ThanePatrol commented on pkgs/development/python-modules/scancode-toolkit/default.nix:
```diff
@@ -130,12 +131,15 @@ buildPythonPackage rec {
+
   nativeCheckInputs = [ pytestCheckHook ];
 
   # Importing scancode needs a writeable home, and preCheck happens in between
```
I can see you've deleted these comments in another commit but for ease of review, please delete comments in the same commit that is logically related to code removal

ThanePatrol reviewed (approved):
LGTM with nits

jopejoe1 committed:
python3Packages.scancode-toolkit: make use of writableTmpDirAsHomeHook

jopejoe1 committed:
python3Packages.scancode-toolkit: pre-generate index file

this is needed to run the scancode cli to scan a codebase otherwise it complains about a non-writable file.

jopejoe1 committed:
python3Packages.scancode-toolkit: renable tests
